src/use-extra-deps/index.ts

```diff
@@ -1,30 +1,29 @@
 /* eslint @typescript-eslint/no-explicit-any: 0 */
 import mapValues from 'lodash/mapValues'
-import omitBy from 'lodash/omitBy'
-import pickBy from 'lodash/pickBy'
-import values from 'lodash/values'
 import * as React from 'react'
 
 // Dependencies that are safe to use in the normal `useEffect` deps array
-export type PrimitiveDep = boolean | string | number | null | void | symbol
+export type PrimitiveDep = boolean
+  | string
+  | number
+  | null
+  | void
+  | symbol
+  | CallbackFn<(...v: any) => any>
 
+declare const __brand: unique symbol
 // Wrapper around a function that has been wrapped in `useSafeCallback`. This
 // type is here to avoid cyclical dependencies.
-export type CallbackFn<F> = {callback: F}
+export type CallbackFn<F> = F & { [__brand]: "CallbackFn" }
```

Comment:
Here's the so called "branding": https://blog.logrocket.com/leveraging-typescript-branded-types-stronger-type-checks/

Reply:
Oh so cool. So basically just extend the thing a bit so it no longer unifies with `F`.